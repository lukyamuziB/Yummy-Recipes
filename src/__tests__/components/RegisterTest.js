import React from 'react';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ReactDom from 'react-dom';
import configureMockStore from 'redux-mock-store';
import { MemoryRouter } from 'react-router-dom';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

import RegisterUser from '../../components/registerUser';

Enzyme.configure({ adapter: new Adapter() });

describe('<RegisterUser />', () => {
  const div = document.createElement('div');
  const store = configureMockStore([thunk])({
    user: {},
  });
  const props = {
    registerUser: jest.fn(() => Promise.resolve('register')),
    validata: jest.fn(),
    onSubmit: jest.fn(),
    history: { push: jest.fn() },
    match: {
      param: {
        id: 1,
      },
    },
  };
  it('should render without crashing', () => {
    <Provider store={store}>
      <MemoryRouter>
        <RegisterUser {...props} />
      </MemoryRouter>
    </Provider>;
  });
  it('renders fully', () => {
    const tree = renderer.create(
      mount(
        <Provider store={store}>
          <MemoryRouter>
            <RegisterUser {...props} />
          </MemoryRouter>
        </Provider>,
      ),
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
