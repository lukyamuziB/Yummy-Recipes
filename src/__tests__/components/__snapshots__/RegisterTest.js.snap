// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegisterUser /> renders fully 1`] = `
<div
  className="landing-container"
>
  <div
    className="in-container"
  >
    <div
      className="inner"
    >
      <div
        className="container"
      >
        <div
          className="container"
        >
          <h5>
             Register Here
          </h5>
           
          <br />
          <form
            onSubmit={[Function]}
          >
            <div
              className="form-group %{touched && error ? 'has-dangetr':''}"
            >
              <lable>
                Name
              </lable>
              <input
                className="form-control"
                name="name"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                type="text"
                value=""
              />
              
            </div>
            <div
              className="form-group %{touched && error ? 'has-dangetr':''}"
            >
              <lable>
                Username
              </lable>
              <input
                className="form-control"
                name="username"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                type="text"
                value=""
              />
              
            </div>
            <div
              className="form-group %{touched && error ? 'has-dangetr':''}"
            >
              <lable>
                Email
              </lable>
              <input
                className="form-control"
                name="email"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                type="email"
                value=""
              />
              
            </div>
            <div
              className="form-group %{touched && error ? 'has-dangetr':''}"
            >
              <lable>
                Password
              </lable>
              <input
                className="form-control"
                name="password"
                onBlur={[Function]}
                onChange={[Function]}
                onDragStart={[Function]}
                onDrop={[Function]}
                onFocus={[Function]}
                type="password"
                value=""
              />
              
            </div>
            <button
              className="btn btn-primary"
              type="submit"
            >
              Submit
            </button>
            <br />
             Alreday Have an Account?
            <a
              href="/login"
              onClick={[Function]}
            >
               Login Here 
            </a>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
